tests/test_datasets/test_hierarchical_dataset.py

```diff
@@ -481,3 +481,11 @@ def test_get_level_dataset_lower_level_error(simple_hierarchical_ts):
 def test_get_level_dataset_with_quantiles(product_level_constant_forecast_w_quantiles, target_level, answer):
     forecast = product_level_constant_forecast_w_quantiles
     np.testing.assert_array_almost_equal(forecast.get_level_dataset(target_level=target_level).df.values, answer)
+
+
+def test_get_level_dataset_pass_target_components_to_output(simple_hierarchical_ts):
```

Comment:
Can you please explain what is happening here and why?

Reply:
Here I test that get_level_dataset pass target_components to the output dataset
It should be done as get_level_dataset creates new dataset with the same columns but different set of segments